### PR TITLE
Replace literal Unicode characters in source code with hex sequence

### PR DESCRIPTION
**Pull request by uklotzde**
https://mixxx.discourse.group/t/windows-build-mixxx/21006

### CODE REVIEW

**Comment by Holzhaus** [comment]
Wait, so MSVC doesn't support Unicode?

Meanwhile in gcc-Land:

Unicode

**Comment by uklotzde** [comment]
Remeber, this is C/C++ code. It is perfectly legal in Rust were source files are by definition UTF-8.

**Comment by Be-ing** [comment]
https://docs.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=msvc-160

**Comment by uklotzde** [comment]
I don't want to explain each and every user how to correctly configure their system. If we can fix it once and for everyone, then let's do it.

**Comment by uklotzde** [comment]
Applies to source code as well as to APIs: https://en.wikipedia.org/wiki/Robustness_principle

**Comment by Be-ing** [comment]
But could we set the compiler flag for MSVC so we don't have to tell people not to put unicode characters in source code?

**Comment by Holzhaus** [comment]
We can add a pre-commit check for that.

**Comment by Be-ing** [comment]
Why get rid of something useful when we could set a compiler flag to make it work?

**Comment by Be-ing** [comment]
I'd still prefer #3563 but I don't care much either way. Merge this if you want.

**Comment by uklotzde** [comment]
We should merge both PRs.

Searching the code for some ASCII string is easier than for UTF-8 symbols that are not available on the keyboard. There are also a lot of different Unicode symbols that almost look identical and could easily be confused.

Never underestimate the creativity of coders :laughing:
https://www.reddit.com/r/rust/comments/5penft/parallelizing_enjarify_in_go_and_rust/dcsgk7n/

File: src/track/keyutils.cpp [hunk 1]
```diff
@@ -7,9 +7,6 @@
 #include "track/keyutils.h"
 #include "util/math.h"
 
-#define MUSIC_FLAT_UTF8  "\xe299ad"
```
Comment (uklotzde) [inline src/track/keyutils.cpp:10]: These constants are actually broken and produce invalid traditional key notation file tags.

**Comment by uklotzde** [comment]
This PR also fixes a real bug as I just noticed.

**Comment by uklotzde** [comment]
Still issues. The traditional key "D♭" is not written correctly into VorbisComment tags.

**Comment by uklotzde** [comment]
The observation regarding key text is unrelated. Weird things happen if you copy&paste Unicode text. I have no clue what is going on. Maybe a Qt issue.

**Comment by uklotzde** [comment]
Suddenly the text field starts with `\u0000` or `\u0002` or `\u0004`, sometimes followed by the Unicode characters `D♭`.

File: src/track/keyutils.cpp [hunk 2]
```diff
@@ -24,39 +21,39 @@ static const QString s_lancelotKeyPattern("^\\s*(1[0-2]|[1-9])([ab])\\s*$");
 // anchor the pattern so we don't get accidental sub-string matches
 // (?:or)? allows unabbreviated major|minor without capturing
 static const QString s_keyPattern = QString::fromUtf8(
-        "^\\s*([a-g])([#♯b♭]*)"
+        "^\\s*([a-g])([#" MUSIC_SHARP_SIGN_UTF8 "b" MUSIC_FLAT_SIGN_UTF8
```
Comment (Holzhaus) [inline src/track/keyutils.cpp:24]: Could this be the issue? If you use `\uXXXX` notation in a regex, doesn't the `\` need to be escaped?

Comment (uklotzde) [reply src/track/keyutils.cpp:24]: Good catch. First we need a regression test.

Comment (uklotzde) [reply src/track/keyutils.cpp:24]: Nope. We already have test for the sharp/flat characters and they succeed. There are no escapes in these strings. Plain UTF-8 that is converted to UTF-16.

Comment (uklotzde) [reply src/track/keyutils.cpp:24]: The expressions are equal. It doesn't matter if you compose them from an UTF-8 character or the corresponding byte sequence.

Comment (uklotzde) [reply src/track/keyutils.cpp:24]: The issues with editing traditional keys with Unicode characters are already present in the current version without this PR. Unrelated.

**Pull request closed without merging**